src/checked.rs

```diff
@@ -56,3 +59,76 @@ impl<T> From<Checked<T>> for Option<T> {
         checked.0.into()
     }
 }
+
+#[cfg(feature = "serde")]
+#[cfg_attr(docsrs, doc(cfg(feature = "serde")))]
+impl<'de, T: Default + Deserialize<'de>> Deserialize<'de> for Checked<T> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = Option::<T>::deserialize(deserializer)?;
+
+        Ok(Self(match value {
+            Some(value) => CtOption::new(value, Choice::from(1)),
+            None => CtOption::new(T::default(), Choice::from(0)),
+        }))
```

Comment:
Perhaps this (provided I got my parens right):

```suggestion
        let value = Option::<T>::deserialize(deserializer)?;
        Self(CtOption::new(value.unwrap_or_default(), Choice::from(value.is_some() as u8)))
```

Reply:
Done, had to make a temporary variable for `Choice` to avoid cloning `value`.